runtime/service.go

```diff
@@ -647,12 +648,16 @@ func (s *service) buildVMConfiguration(req *proto.CreateVMRequest) (*firecracker
 
 	var driveBuilder firecracker.DrivesBuilder
 	// Create non-stub drives
+	var path = s.config.RootDrive
```

Comment:
Do we want the configuration `RootDrive` field to be read only? They may want a read/write root drive. Either we make it read/write or introduce a `RootDriveIsReadOnly` or something in the configuration.

Reply:
We likely want the root drive to be read-only by default.  Running multiple VMs from the same root drive where it's mutable can corrupt the drive; multiple writers to the same disk isn't really supported by common filesystems.  The image builder that we have in this repo will generate a VM that runs from a read-only root filesystem, but then mounts a read-write filesystem on top with overlay so that the VM thinks it's writable.

Reply:
Customers can have different files by utilizing the image builder tool. There might be a customer who want to put files at runtime, but supporting that is possible by what @xibz suggested (adding that in a request and/or the configuration file).

However, since we don't have the customer request, we can start from the safest option -- always read-only.

Reply:
Actually, as @sipsma pointed out, CrateVMRequest is already having a field to make the rootfs read-only, which we are discarding even before my change.

I've fixed that in the last revision. So the default is read-only but customers can override that through CrateVMRequest. The JSON configuration doesn't have the way to make the rootfs writable though.